Add version info when building CI/CD for apple platforms

kiddkaffeine opened the pull request:
Sets minimum OS version when building for macOS, iOS, and tvOS
See: https://github.com/libsdl-org/SDL/issues/11332 and https://github.com/libsdl-org/SDL/issues/11333

## Description
This sets the minimum version for the following platforms rather than using the newest:
macOS to 11.0 or higher (currently 14)
iOS to 13.0 or higher (currently 17)
tvOS to 13.0 or higher (currently 17)
Using the values suggested at https://github.com/libsdl-org/SDL/issues/11332#issuecomment-2438953930

## Existing Issue(s)
This has revealed a few issues with SDL GPU on older iOS/tvOS versions; already discussed in https://github.com/libsdl-org/SDL/issues/11333 and directly with TheSpyDog

maia-s commented:
I know 11 is required for macos on aarch64, but is it also required for macos on x86-64? I ask because I know some people want to stay on 10.14 because it's the last version of macos with support for 32-bit apps. If it could be reduced to 10.13 or 10.11 it'd support more earlier macs, too

slouken commented:
These are the official SDL deployment targets:
IPHONEOS_DEPLOYMENT_TARGET = 9.0;
MACOSX_DEPLOYMENT_TARGET = 10.11;
TVOS_DEPLOYMENT_TARGET = 9.0;
XROS_DEPLOYMENT_TARGET = 1.0;

These are earlier than the versions supported by Xcode, but they currently still work, albeit with warnings. We would accept patches to fix warnings, but not to change the deployment versions at this time.

kiddkaffeine commented:
> I know 11 is required for macos on aarch64, but is it also required for macos on x86-64? I ask because I know some people want to stay on 10.14 because it's the last version of macos with support for 32-bit apps. If it could be reduced to 10.13 or 10.11 it'd support more earlier macs, too

So just to point out, I don't think there's any reason you couldn't compile for earlier versions, although some subsystems may be unavailable on those older versions.

The reason I submitted this PR is that the SDL CI/CD is building solely against the current Mac/iOS/tvOS API, which means that...
1) API calls that do not work on older macOS/iOS/tvOS versions may go unnoticed, and
2) The binaries built currently can crash on anything not within a version or two of latest.

I have adjusted the version numbers as mentioned by @slouken above, but the CI/CD treats some of these version warnings as errors, and a few components (notably SDL_GPU) are probably going to need some runtime checks.  CC @TheSpydog

madebr commented:
I think these these belong in the cmake script, before the first target (SDL3-shared or SDL3-static) is created.

madebr commented on CMakeLists.txt:
```diff
@@ -4,6 +4,15 @@ if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_CURRENT_BINARY_DIR)
 endif()
 
+# This needs to be set before project() calls
+if(CMAKE_SYSTEM_NAME MATCHES "iOS" OR CMAKE_SYSTEM_NAME MATCHES "tvOS")
```
If you move these under the call to `SDL_DetectCMakePlatform()`, the cmake script defines `TVOS`, `IOS`, `VISIONOS` and `MACOS` CMake variables.

kiddkaffeine commented:
as noted there are some issues doing this inside the CMake file, but on further inspection, the sibling variable (CMAKE_OSX_ARCHITECTURES) is set on the command line, and SDL's instructions even say to set this on the command line.

Therefore, I've reverted CMAKE_OSX_DEPLOYMENT_TARGET to match the use of CMAKE_OSX_ARCHITECTURES, and added this to the documentation.

kiddkaffeine commented:
All Apple platforms are currently green with @slouken 's requested targets, so this is the last piece remaining currently.
